File: C++/0273.cpp
#include <iostream>
using namespace std;


int main()
{
    char num[101];
    bool was[1000] = { 0 };
    int cnt = 0;

    cin >> num;

    for (int d1 = 0; num[d1] != '\0'; d1++)
    {
        for (int d2 = d1 + 1; num[d2] != '\0'; d2++)
        {
            for (int d3 = d2 + 1; num[d3] != '\0'; d3++)
            {
                int digit1 = (num[d1] - '0');
                int digit2 = (num[d2] - '0');
                int digit3 = (num[d3] - '0');
                was[digit1 * 100 + digit2 * 10 + digit3] = true;
            }
        }
    }

    for (int i = 100; i <= 999; i++)
        cnt += was[i];

    cout << cnt << endl;

    return 0;
}

File: Python/0273.py
num = input()
was = [0] * 1000

for d1 in range(len(num) - 2):
    for d2 in range(d1 + 1, len(num) - 1):
        for d3 in range(d2 + 1, len(num)):
            was[int(num[d1]) * 100 + int(num[d2]) * 10 + int(num[d3])] = 1

cnt = 0
for i in range(100, 1000):
    cnt += was[i]

print(cnt)
